File: Fifo-Match-Application-main/Fifo-Match-Application-main/app/src/main/java/com/fifo/match/model/NotificationModel.kt
package com.fifo.match.model


import com.google.gson.annotations.SerializedName

data class NotificationModel(
    @SerializedName("id")
    val id: Int?,
    @SerializedName("from_id")
    val fromId: Int?,
    @SerializedName("to_id")
    val toId: Int?,
    @SerializedName("open_id")
    val openId: Int?,
    @SerializedName("title")
    val title: String?,
    @SerializedName("subtitle")
    val subtitle: String?,
    @SerializedName("type")
    val type: String?,
    @SerializedName("is_read")
    val isRead: Boolean?,
    @SerializedName("created_at")
    val createdAt: String?,
    @SerializedName("updated_at")
    val updatedAt: String?,
    @SerializedName("deleted_at")
    val deletedAt: Any?,
    @SerializedName("photo")
    val photo: Photo?
) {

    data class Photo(
        @SerializedName("id")
        val id: Int?,
        @SerializedName("user_id")
        val userId: Int?,
        @SerializedName("type")
        val type: Int?,
        @SerializedName("name")
        val name: String?,
        @SerializedName("created_at")
        val createdAt: String?,
        @SerializedName("updated_at")
        val updatedAt: String?,
        @SerializedName("deleted_at")
        val deletedAt: Any?
    )
}

File: Fifo-Match-IOS-main/Fifo-Match-IOS-main/FIFOMatch/Models/Notifications/NotificationModel.swift
//
//  NotificationModel.swift
//  FIFOMatch
//
//  Created by Harendra Singh Rathore on 01/04/22.
//

import Foundation

// MARK: - NotificationModel
struct NotificationModel: Decodable {
  let msg, message: String?
  let status: Int?
  let data: [NotificationData]?
}

// MARK: - NotificationData
struct NotificationData: Decodable {
  let id, fromID, toID: Int?
  let title, subtitle: String?
  let createdAt, updatedAt: String?
  let isRead: Bool?
  let photo: UserPhoto?
  let type: String?
  let openId: Int?
  
  enum CodingKeys: String, CodingKey {
    case id
    case fromID = "from_id"
    case toID = "to_id"
    case title, subtitle
    case isRead = "is_read"
    case createdAt = "created_at"
    case updatedAt = "updated_at"
    case photo
    case type
    case openId = "open_id"
  }
}

// MARK: - NotificationModel
struct ClearNotificationModel: Decodable {
  let msg, message: String?
  let status: Int?
}
